pkg/middleware/auth.go

```diff
@@ -181,6 +281,23 @@ func (a Auth) AuthorizeAPI(next http.Handler) http.Handler {
 // NewAuth makes Auth middleware
 func NewAuth(authConfig *config.ServiceAuthConfig) Auth {
 
+	if authConfig.JwksURL != "" && authConfig.HMACSecret != "" {
+		log.Warn().Msg("HMAC Secret will be ignored, JWKS URL will be used for token validation")
+	}
+
+	if authConfig.JwksURL != "" {
+		if authConfig.JwksUpdateInterval <= 0 {
+			log.Error().Msg("JwksUpdateInterval must be set")
+			return Auth{Verifier: BadAuth{}}
+		}
+		verifier := NewJWTVerifierURL(authConfig.JwksURL, authConfig.JwksUpdateInterval)
+		if verifier.jwksKeys == nil {
```

Comment:
Instead of `verifier.jwksKeys == nil`, perhaps: `verifier == nil` or `verifier, err := NewJWTVerifierURL(..); err != nil` would be clearer.

Reply:
verifier is the interface, I was returning everything by value for other cases. NewJWTVerifierURL needs to return by pointer because of locking logic. I can make other cases consistent, return by pointer, and then remove BadAuth.
`NewAuth(authConfig *config.ServiceAuthConfig) Auth` will need to become `NewAuth(authConfig *config.ServiceAuthConfig) *Auth` .